.deepsource.toml

```diff
@@ -30,6 +31,7 @@ name = "python"
 enabled = true
 dependency_file_paths = [
   "requirements.txt",
+  "tests/requirements.txt",
```

Comment:
This has already been done. CI does use 'tests/requirements.txt':
* please read .github/workflows/Tests.yml (ie one of the configs for CI) Note the use of `pip ... -r tests/requirements.txt` (for example: https://github.com/reactive-firewall/multicast/blob/master/.github/workflows/Tests.yml#L156-L156 as part of the minimal acceptance tests (aka MATs))

* what are you considering "the CI workflow" ? elaborate so I may clear up any misunderstandings on what is and is-not the CI workflow for this project.

However there is additional opportunity:
* please read .circleci/config.yml (ie one of the other configs for CI) and consider we could improve consistency around [line 120](https://github.com/reactive-firewall/multicast/blob/master/.circleci/config.yml#L120-L143) by instead using 'tests/requirements.txt'
* we can also explore adding a specific test for verifying the 'tests/requirements.txt' file itself :shrug:

Reply:
Regarding your difficulties finding the use of 'tests/requirements.txt' in '.github/workflows/Tests.yml' I think I see where you are confused:
* First here are the places where 'tests/requirements.txt' is used:
  [Line 156](https://github.com/reactive-firewall/multicast/blob/2a7719b28299c44e6406866f10ad8c8b1fdc5fbf/.github/workflows/Tests.yml#L156-L156), [Line 204](https://github.com/reactive-firewall/multicast/blob/2a7719b28299c44e6406866f10ad8c8b1fdc5fbf/.github/workflows/Tests.yml#L204-L204), [Line 286](https://github.com/reactive-firewall/multicast/blob/2a7719b28299c44e6406866f10ad8c8b1fdc5fbf/.github/workflows/Tests.yml#L286-L286), [Line 360](https://github.com/reactive-firewall/multicast/blob/2a7719b28299c44e6406866f10ad8c8b1fdc5fbf/.github/workflows/Tests.yml#L360-L360), [Line 404](https://github.com/reactive-firewall/multicast/blob/2a7719b28299c44e6406866f10ad8c8b1fdc5fbf/.github/workflows/Tests.yml#L404-L404), [Line 501](https://github.com/reactive-firewall/multicast/blob/2a7719b28299c44e6406866f10ad8c8b1fdc5fbf/.github/workflows/Tests.yml#L501-L501) and [Line 566](https://github.com/reactive-firewall/multicast/blob/2a7719b28299c44e6406866f10ad8c8b1fdc5fbf/.github/workflows/Tests.yml#L566-566)
* Notice the reference to the path: `./tests/requirements.txt`. This of-course is the same as 'tests/requirements.txt' with the disambiguating prefix of dot-slash.
* I suspect if you just expanded your search to optionally match the dot-slash `./` prefix you would not have missed it.In the future when you can't find any file-path occurrences to an exact match, you can follow-up by exploring searching for the same file-path prefixed by dot-slash "./" (even the recursion of this can be confusing when written as '././'). Also if you do find a match this way you can verify it with the use of the `readlink` tool and its `-f` argument but that is off-topic.

* I look forward to what you make of this information. 
  
  * 🐰 us humans would note your oversight of line 404 is ironic because like a http 404 error you did not find what you were looking for on line 404. Of-course irony is not a KPI for code-reviews.

Reply:
now, let's focus on exploring the idea of adding a specific test for verifying the 'tests/requirements.txt' file itself

Reply:
🤩 These are almost perfect. I especially appreciate the multifaceted approach, very robust.
* Regarding 4, this is partially implemented, and the CircleCI part can be planned as part of #60 chores. 
* Regarding 6 this is actually implemented already, please read '.github/dependabot.yml' and you'll find it implemented around line 44. Although I used the value "tests/" for the 'directory' key, which should be the same "tests", unless I'm mistaken about the dpendabot.yml docs saying something about the 'directory' key's value being relative to the project's root.

* So, Let's focus on 1, 2, 3, and 5:
 - concerning 5, please open a new GH issue for now and we'll circle back to it later 
 - concerning 1, 2, & 3
   * please also open another a new GH issue to track the work.
   * we can probably group these related testcases together during implementation
   * we'll need to explore various consistency improvements